src/editable-heading/editable-heading.tsx

```diff
@@ -121,15 +130,33 @@ export const EditableHeading = (props: EditableHeadingProps) => {
     setIsInSelectionMode(false);
   }, [isMouseDown, isInSelectionMode, disabled, onEdit]);
 
-  const onInputFocus = React.useCallback((e: React.FocusEvent<HTMLInputElement>) => {
-    setIsInFocus(true);
-    onFocus?.(e);
-  }, [onFocus]);
-
-  const onInputBlur = React.useCallback((e: React.FocusEvent<HTMLInputElement>) => {
-    setIsInFocus(false);
-    onBlur?.(e);
-  }, [onBlur]);
+  const onInputFocus = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => {
+      setIsInFocus(true);
+      onFocus?.(e);
+    }, [onFocus]);
+
+  const onInputBlur = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => {
+      setIsInFocus(false);
+      onBlur?.(e);
+    }, [onBlur]);
+
+  textAreaRef?.current?.addEventListener('scroll', () => {
+    const scrollHeight = textAreaRef?.current?.scrollHeight || 0;
```

Comment:
I believe, first optional chaining is redundant (`textAreaRef?.`)

also, wouldn't it be simplier to just have 1 check in the beginning?|
```
if (!textAreaRef.current) {
  return
}
```

Reply:
Unfortunately we can't apply that rule here. Because in my case we just won't hover lisner if we don't have ref on textarea. In your case, we will do a full return and won't render anything if we don't have a ref, which will break the component. Attached screenshot
<img width="1920" alt="Screenshot 2023-11-04 at 19 12 32" src="https://github.com/JetBrains/ring-ui/assets/10141840/2352887c-d96e-4ebc-8cf8-264460472900">

Reply:
Is this comment should resolve or maybe I have something missed?

Reply:
Yep, thanks